StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift

```diff
@@ -424,108 +433,124 @@ extension LinkAccountPickerViewController: LinkAccountPickerBodyViewDelegate {
+                        guard let self else { return }
+                        if selectedPartnerAccount.nextPaneOnSelection == .partnerAuth {
+                            if let institution = selectedPartnerAccount.institution {
+                                self.delegate?.linkAccountPickerViewController(
+                                    self,
+                                    requestedPartnerAuthWithInstitution: institution
+                                )
+                            } else {
+                                self.delegate?.linkAccountPickerViewController(
+                                    self,
+                                    didRequestNextPane: .institutionPicker
+                                )
+                            }
+                        }
+                        // nextPaneOnSelection == bankAuthRepair
+                        else {
+                            dataSource
+                                .analyticsClient
+                                .logUnexpectedError(
+                                    FinancialConnectionsSheetError
+                                        .unknown(
+                                            debugDescription: "Updating a repair account, but repairs are not supported in Mobile."
+                                        ),
+                                    errorName: "UpdateRepairAccountError",
+                                    pane: .linkAccountPicker
+                                )
+                            delegate?.linkAccountPickerViewController(
                                 self,
                                 didRequestNextPane: .institutionPicker
                             )
                         }
                     }
-                    // nextPaneOnSelection == bankAuthRepair
-                    else {
-                        dataSource
-                            .analyticsClient
-                            .logUnexpectedError(
-                                FinancialConnectionsSheetError
-                                    .unknown(
-                                        debugDescription: "Updating a repair account, but repairs are not supported in Mobile."
-                                    ),
-                                errorName: "UpdateRepairAccountError",
-                                pane: .linkAccountPicker
+
+                    let genericInfoViewController = GenericInfoViewController(
+                        genericInfoScreen: drawerOnSelection,
+                        theme: dataSource.manifest.theme,
+                        panePresentationStyle: .sheet,
+                        iconView: {
+                            let institutionIconView = InstitutionIconView()
+                            institutionIconView.setImageUrl(selectedPartnerAccount.institution?.icon?.default)
+                            return institutionIconView
+                        }(),
```

Comment:
we override the backend icon because the icon is dependent on selection (and backend doesn't know the selection):

[Stripe.js override](https://git.corp.stripe.com/stripe-internal/stripe-js-v3/pull/23288/files#diff-b2beff986072eb286873f4aea4307fedbe4b05a9c4b1e5328fb1cded1253b1f3R35)

Reply:
What happens here if `selectedPartnerAccount.institution?.icon?.default` is nil? Should we maybe only return the `institutionIconView` here if that institution icon isn't nil?

Reply:
ty improved here:

https://github.com/stripe/stripe-ios/pull/3833/commits/dc24a89977bceada6a613d95ce34e3c14f1b3560